discovery/ovhcloud/ovhcloud_test.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ovhcloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
```

Comment:
I wonder if it is worth it to introduce this dependency in Prometheus.

Reply:
We feel like it could be a nice addition for the test suite, for the code we wrote but also for the upcoming features from the others contributors. If you think it is not worth it, we can remove it.

Reply:
I would like to avoid extra dependencies like this.

Service discovery is also used by third parties as libraries, so it impacts way more than Prometheus.

Reply:
In my opinion, third parties doesn't import test files so they are not affected by libraries on _test.go files